### PR TITLE
React to aspnet/KestrelHttpServer#2496: make IConnectionInherentKeepAliveFeature a boolean feature

### PR DESCRIPTION
**Pull request by anurse**
Reaction to aspnet/KestrelHttpServer#2496

### CODE REVIEW

File: benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubConnectionStartBenchmark.cs [hunk 1]
```diff
@@ -76,6 +76,6 @@ private void AddHandshakeResponse()
 
     public class TestConnectionInherentKeepAliveFeature : IConnectionInherentKeepAliveFeature
     {
-        public TimeSpan KeepAliveInterval { get; } = TimeSpan.Zero;
+        public bool HasInherentKeepAlive => true;
```
Comment (anurse) [inline benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubConnectionStartBenchmark.cs]: Fixing

File: samples/ClientSample/Tcp/TcpConnection.cs [hunk 2]
```diff
@@ -40,7 +40,8 @@ public TcpConnection(EndPoint endPoint)
         public override string ConnectionId { get; set; } = Guid.NewGuid().ToString();
         public override IDictionary<object, object> Items { get; set; } = new ConnectionItems();
 
-        public TimeSpan KeepAliveInterval => Timeout.InfiniteTimeSpan;
+        // We claim to have inherent keep-alive so the client doesn't kill the connection when it hasn't seen ping frames.
+        public bool HasInherentKeepAlive => true;
```
Comment (anurse) [inline samples/ClientSample/Tcp/TcpConnection.cs]: Fixing

File: src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs [hunk 3]
```diff
@@ -60,6 +59,9 @@ public override IDuplexPipe Transport
         public override string ConnectionId { get; set; }
         public override IDictionary<object, object> Items { get; set; } = new ConnectionItems();
 
+        // REVIEW: Public? I can private-impl it instead. HubConnection uses ConnectionContext.Features to get this info.
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs]: Remove comment, public is dine.

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 4]
```diff
@@ -76,6 +76,9 @@ public EndToEndTests(ServerFixture<Startup> serverFixture, ITestOutputHelper out
         [MemberData(nameof(TransportTypes))]
         public async Task CanStartAndStopConnectionUsingGivenTransport(HttpTransportType transportType)
         {
+            const int TaskCount = 100;
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: uh?

Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: derp. other change got mixed in

Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: well, other investigation

**Pull request merged**